pkg/waiter/manager.go

```diff
@@ -0,0 +1,255 @@
+package waiter
+
+import (
+	"context"
+	"errors"
+
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+const (
+	// The default number of attempts to re-generate an ID on collision.
+	defaultMaxCollisions = 5
```

Comment:
Certainly a debated topic :).  It may have a low chance of collision, but a collision is still possible - handling the collision is simple whereas being in a situation where a collision occurs at runtime can cause major issues. 

I'll remove the ability to configure the number of collisions and just use the `const` to simplify things a bit.  My preference would be the keep the collision logic however, unless you feel strongly about it.  Let me know if you do.